### PR TITLE
Adding an expense being overbudget now shows a notification instead of a warning

### PR DESCRIPTION
**Pull request by winsonhys**

### CODE REVIEW

File: src/main/java/seedu/address/model/budget/TotalBudget.java [hunk 1]
```diff
@@ -130,6 +135,7 @@ public void setCategoryBudget(CategoryBudget budget) throws CategoryBudgetExceed
 
     @Override
     public void clearSpending() {
+        System.out.println("Clear spending is called");
```
Comment (Snookerballs) [inline src/main/java/seedu/address/model/budget/TotalBudget.java]: Remove Debug Statements

File: src/main/java/seedu/address/model/budget/TotalBudget.java [hunk 2]
```diff
@@ -18,11 +18,16 @@
  */
 public class TotalBudget extends Budget {
 
+    public static final String NOT_SET = "NOT_SET";
+    public static final String SPENDING_RESET = "SPENDING_RESET";
+    public static final String DO_NOTHING = "DO_NOTHING";
+
     protected long numberOfSecondsToRecurAgain;
     private LocalDateTime nextRecurrence;
     private HashSet<CategoryBudget> categoryBudgets;
```
Comment (Snookerballs) [inline src/main/java/seedu/address/model/budget/TotalBudget.java]: Remove noise

File: src/test/java/seedu/address/logic/commands/BudgetIntegrationTest.java [hunk 3]
```diff
@@ -39,6 +41,21 @@ public void addExpense_budgetUpdated() throws CommandException, NoUserSelectedEx
...
+        Category selectedCategory = model.getFilteredExpenseList().get(0).getCategory();
+        double totalExpensesOfSelectedCategory =
+            model.getFilteredExpenseList().stream().filter(x -> x.getCategory().equals(selectedCategory))
+                .mapToDouble(x -> x.getCost().getCostValue()).sum();
+        this.model.modifyMaximumBudget(new TotalBudget("99999.00"));
+        new SetCategoryBudgetCommand(new CategoryBudget(selectedCategory.categoryName, "999.00"))
+            .execute(this.model, this.commandHistory);
+        CategoryBudget cBudget = this.model.getMaximumBudget().getCategoryBudgets().toArray(new CategoryBudget[1])[0];
+        assertEquals(cBudget.getCurrentExpenses(), totalExpensesOfSelectedCategory);
+
```
Comment (Snookerballs) [inline src/test/java/seedu/address/logic/commands/BudgetIntegrationTest.java]: noise

File: src/main/java/seedu/address/model/budget/Budget.java [hunk 4]
```diff
@@ -49,6 +49,15 @@ public Budget (String budget) {
 
     }
 
+    public Budget (String budget, String currentExpenses) {
+        requireNonNull(budget);
+        checkArgument(isValidBudget(budget), BUDGET_VALIDATION_REGEX);
+        checkArgument(isValidBudget(currentExpenses), BUDGET_VALIDATION_REGEX);
+        this.budgetCap = Double.parseDouble(budget);
+        this.currentExpenses = Double.parseDouble(currentExpenses);
+
```
Comment (Snookerballs) [inline src/main/java/seedu/address/model/budget/Budget.java]: noise

**Comment by winsonhys** [comment]
Things you commented "noise" in the PRs are not noise. Noise happens when you make unnecessary changes to new files without changing the structure, logic nor documentation of the code, ie a deletion/insertion of a new line in a file and no other edits were made in the same file.

**Review by Snookerballs — APPROVED** [review]

**Pull request merged**